### PR TITLE
build: Support Python 3.7+

### LINKED ISSUE #244: Require python 3.7 or better
We are using `backoff` > 2.0, and that requires 3.6 or better - so we should probably match so it doesn't look like false advertising.

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #244

* Require Python 3.7+ and remove upper cap on `python_requires`
   - c.f. https://iscinumpy.dev/post/bound-version-constraints/#pinning-the-python-version-is-special
* Update classifier metadata on PyPI.
* Update Python versions in README and elsewhere.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@gordonwatts this is ready for review. I'm going to have a few PRs before I do docs that will make it easier to do them in general.

File: setup.py [hunk 1]
```diff
@@ -8,9 +8,6 @@
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
-if sys.version_info[0] < 3:
-    raise NotImplementedError("Do not support version 2 of python")
-
```
Comment (matthewfeickert) [inline setup.py:13]: `python_requires` already strictly enforces this, so this is essentially dead code so remove it.

**Comment by matthewfeickert** [comment]
@gordonwatts now that PR #245 is in it is reasonable over the next `n` days to start reviewing the PRs that will lead to PR #249? Or do you and @talvandaalen have some additional work that you wanted to get in before that?

**Comment by gordonwatts** [comment]
Yeah - @talvandaalen needs to do some integration testing first.

**Comment by matthewfeickert** [comment]
> Yeah - @talvandaalen needs to do some integration testing first.

Okay if that's all then PR #246 (this one), PR #247, and PR #248 can all be reviewed independently of @talvandaalen's work as they just deal with setting up infrastructure / tooling to make testing and docs management easier &mdash; they don't touch anything in the `servicex/` directory.

I can then just keep rebasing PR #249 along with @talvandaalen's work to be able to ensure that the API references (once the other Sphinx stuff is cleaned up more) are all looking good in the docs.

**Comment by matthewfeickert** [comment]
Thanks for the merge @gordonwatts. I'll start rebasing my other PRs.

**Pull request merged**